web/wp-content/plugins/tmbi-adobe-dtm/inc/rd-dtm-settings.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: jeysaravana
+ * Date: 2017-01-20
+ * Time: 12:21 PM
+ */
+
+class RD_DTM_Settings {
+
+	const RD_DTM_SLUG        = 'rd-adobe-dtm';
+	const NICKNAME_SLUG      = 'rd-dtm-nickname';
+	const AMP_SERVER_ACCOUNT = 'rd-dtm-amp-server';
+	const SETTING_NAME       = 'RD Adobe DTM';
+	private $options = array();
+
+	public function __construct() {
+
+		add_action( 'admin_menu', array( $this, 'admin_settings' ) );
+		add_action( 'admin_init', array( $this, 'admin_page_init' ) );
+		add_action( 'wp_footer', array( $this, 'rd_dtm_footer' ), 100000 );
+
+	}
+
+	public function rd_dtm_footer() {
```

Comment:
Class seems to be for a Settings page, but injects JS in the footer (which can't be dequeued/reprioritized from the theme because `add_action` refers to `$this`).
This function probably belongs to a different class, and should be hooked in a way that doesn't take control away from the frontend layer.

Reply:
See https://readersdigest.atlassian.net/browse/DRDT-102